src/app/settings/security.component.ts

```diff
@@ -0,0 +1,25 @@
+import { Component, OnInit } from "@angular/core";
+import { ActivatedRoute, Router } from "@angular/router";
+
+import { KeyConnectorService } from "jslib-common/abstractions/keyConnector.service";
+
+@Component({
+  selector: "app-security",
+  templateUrl: "security.component.html",
+})
+export class SecurityComponent implements OnInit {
+  showChangePassword = true;
+
+  constructor(
+    private keyConnectorService: KeyConnectorService,
+    private router: Router,
+    private route: ActivatedRoute
+  ) {}
+
+  async ngOnInit() {
+    this.showChangePassword = !(await this.keyConnectorService.getUsesKeyConnector());
+    if (!this.showChangePassword) {
+      this.router.navigate(["/settings/security/two-factor"]);
+    }
+  }
```

Comment:
This could potentially be messy if the user navigates to any security page they will get re-directed to two-factor. Is the goal to hide the change-password page? If so could we move the logic to that page?

Reply:
Not sure I follow. The `security` page's default route is `settings/security/change-password` but there is an instance where the `change-password` tab is not available so the component handles navigating to the next available tab. This only happens when the `security` component is initialized via selection from the left-handed vertical menu. `two-factor` is always present and visible. Since the `security` component handles the available tabs, my thought is that is makes the most sense to handle the above scenario here.

Reply:
This component is the "layout" component for each security page right? If you say navigate directly to `settings/security/security-keys` It will redirect you to `two-factor` since it triggers the `ngOnInit` for the SecurityComponent.

Reply:
Ah! - thanks so much for clarifying. I've moved the navigation reroute into the `change-password` component which will protect against the scenario above as well as someone trying to route directly to `settings/security/change-password`.